metadata/src/main/java/org/apache/kafka/controller/ClusterControlManager.java

```diff
@@ -239,11 +239,11 @@ public void replay(RegisterBrokerRecord record) {
             features.put(feature.name(), new VersionRange(
                 feature.minSupportedVersion(), feature.maxSupportedVersion()));
         }
+        BrokerRegistration prevRegistration = brokerRegistrations.getOrDefault(brokerId, null);
```

Comment:
`Map::put` returns the previous value.

Reply:
This was still a bug then correct? `prevRegistration` was being assigned to `Map::get` not `Map::put`.

Reply:
Yes, it looks like it affects the log message printed, right? It doesn't explain the behavior you are seeing.

Reply:
Yes it affects the logging.